lib/srv/db/mysql/engine.go

```diff
@@ -191,6 +197,26 @@ func (e *Engine) connect(ctx context.Context, sessionCtx *common.Session) (*clie
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		// workaround issue generating ephemeral certs for secure connections
+		// by creating a TLS connection to the cloud proxy port (3307) and
+		// overridding mysql client's connection. mysql on 3306 does not trust
+		// the ephemeral cert's CA but cloud proxy does.
+		if sessionCtx.Database.GetTLS().Mode != types.DatabaseTLSMode_INSECURE {
+			uri := sessionCtx.Database.GetURI()
+			host, port, err := net.SplitHostPort(uri)
+			if err == nil && port == "3306" {
+				uri = net.JoinHostPort(host, "3307")
+				e.Log.Debug("Overrided URI port from 3306 to 3307")
```

Comment:
Nit: Don't know if you intend to keep this log entry in the final implementation but not sure how useful it really is.